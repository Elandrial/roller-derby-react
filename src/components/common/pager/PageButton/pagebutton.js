export default function PageButton({
    currentPage,
    pageNumber,
    ariaName,
    hiddenLG,
    hiddenMD,
    hiddenSM,
    hiddenXS,
    onClick
}){
    let cssClasses = [];
    let ariaLabel;
    let disabled = false;

    if(hiddenLG) cssClasses.push('hidden-l');
    if(hiddenMD) cssClasses.push('hidden-m');
    if(hiddenSM) cssClasses.push('hidden-s');
    if(hiddenXS) cssClasses.push('hidden-xs');


    if(currentPage === pageNumber){
        ariaLabel = `Current ${ariaName} Page, Page ${pageNumber}`;
        disabled = true;
    }
    else{
        ariaLabel = `Go to ${ariaName} page ${pageNumber}`;
    }

    return (
        <button disabled={disabled} 
                aria-label={ariaLabel} 
                aria-current="true" 
                key={pageNumber} 
                className={cssClasses.join(' ')}
                onClick={() => onClick(pageNumber)}>
            {pageNumber}
        </button>
    );
}