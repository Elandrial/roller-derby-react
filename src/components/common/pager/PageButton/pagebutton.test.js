import PageButton from './pagebutton';
import renderer from 'react-test-renderer';

describe('PageButton component',() => {
  it('Page 1, current button renders correctly', () =>{
    const tree = renderer
    .create(<PageButton currentPage={1} pageNumber={1} ariaName="Events" hiddenLG={false} hiddenMD={false} hiddenSM={false} hiddenXS={false}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
  })

  it('Page 2, not current button renders correctly', () =>{
    const tree = renderer
    .create(<PageButton currentPage={1} pageNumber={2} ariaName="Events" hiddenLG={false} hiddenMD={false} hiddenSM={false} hiddenXS={false}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
  })

  it('Page 6, not current button renders correctly, hidden on LG, MD', () =>{
    const tree = renderer
    .create(<PageButton currentPage={1} pageNumber={6} ariaName="Events" hiddenLG={true} hiddenMD={true} hiddenSM={false} hiddenXS={false}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
  })
});