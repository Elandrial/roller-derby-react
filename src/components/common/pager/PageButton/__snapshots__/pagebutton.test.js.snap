// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageButton component Page 1, current button renders correctly 1`] = `
<button
  aria-current="true"
  aria-label="Current Events Page, Page 1"
  className=""
  disabled={true}
  onClick={[Function]}
>
  1
</button>
`;

exports[`PageButton component Page 2, not current button renders correctly 1`] = `
<button
  aria-current="true"
  aria-label="Go to Events page 2"
  className=""
  disabled={false}
  onClick={[Function]}
>
  2
</button>
`;

exports[`PageButton component Page 6, not current button renders correctly, hidden on LG, MD 1`] = `
<button
  aria-current="true"
  aria-label="Go to Events page 6"
  className="hidden-l hidden-m"
  disabled={false}
  onClick={[Function]}
>
  6
</button>
`;
